src/java/test/org/apache/zookeeper/server/quorum/QuorumPeerMainTest.java

```diff
@@ -413,13 +418,18 @@ public void testBadPeerAddressInQuorum() throws Exception {
         ClientBase.setupTestEnv();
 
         // setup the logger to capture all logs
+        LoggerContext loggerContext =  (LoggerContext) LogManager.getContext(false);
```

Comment:
@jvz - I tried to update tests using the `ListAppender` but I don't think, it is the right way to verify the log messages. I observed that when I run all the tests in the `QuorumPeerMainTest` or `ReadOnlyModeTest`, tests fail. However, running 
following test using `test.method=` option works fine 
`QuorumPeerMainTest. testInconsistentPeerType` 
`QuorumPeerMainTest. testQuorumDefaults`
`QuorumPeerMainTest. testBadPeerAddressInQuorum`
`ReadOnlyModeTest.testSeekForRwServer`

Not sure what am I missing here.

Let me know what you think.